.github/workflows/ci.yml

```diff
@@ -402,7 +465,7 @@ jobs:
           flags: cloud
 
   integration-tests:
```

Comment:
I think you need to add:
```        
    with:
        ref: ${{ github.event.pull_request.head.sha }}
```
to the Checkout step in this job.

Reply:
Yep. Adding it here.